dcicutils/env_utils.py

```diff
@@ -415,3 +417,34 @@ def full_fourfront_env_name(envname):
     check_true(isinstance(envname, str) and "cgap" not in envname, "The envname is not a Fourfront env name.",
                error_class=ValueError)
     return full_env_name(envname)
+
+
+def classify_server_url(url, raise_error=True):
+
+    parsed = urlparse(url)
+    hostname = parsed.hostname
+    hostname1 = hostname.split('.', 1)[0]  # The part before the first dot (if any)
+
+    environment = get_bucket_env(hostname1)  # First approximation, maybe overridden below
+
+    is_stg_or_prd = is_stg_or_prd_env(hostname1)
+
+    if hostname1 == 'localhost' or hostname == '127.0.0.1':
+        environment = None
+        kind = 'localhost'
+    elif 'cgap' in hostname1:
+        kind = 'cgap'
+    elif is_stg_or_prd or 'fourfront-' in hostname1:
+        kind = 'fourfront'
+    else:
+        if raise_error:
+            raise RuntimeError("%s is not a Fourfront or CGAP server." % url)
+        else:
+            environment = None
+            kind = None
```

Comment:
Can we have "unknown" for these or something? `None` I don't think will be represented well depending on how this error is reported.

Reply:
yeah, i thought about that and it was an aesthetic choice but not a strong one. Happy to accommodate.